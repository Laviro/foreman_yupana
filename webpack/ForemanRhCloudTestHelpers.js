export const inventoryStateWrapper = innerState => ({
  ForemanRhCloud: {
    inventoryUpload: { ...innerState },
  },
});
