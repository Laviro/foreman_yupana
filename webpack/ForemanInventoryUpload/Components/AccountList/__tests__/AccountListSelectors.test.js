import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';
import {
  selectAccountsList,
  selectAccounts,
  selectPollingProcessID,
  selectAutoUploadEnabled,
  selectHostObfuscationEnabled,
} from '../AccountListSelectors';
import {
  pollingProcessID,
  accounts,
  autoUploadEnabled,
  hostObfuscationEnabled,
} from '../AccountList.fixtures';
import { rhCloudStateWrapper } from '../../../../ForemanRhCloudTestHelpers';

const state = rhCloudStateWrapper({
  accountsList: {
    accounts,
    pollingProcessID,
    autoUploadEnabled,
    hostObfuscationEnabled,
  },
});

const fixtures = {
  'should return AccountsList': () => selectAccountsList(state),
  'should return AccountList accounts': () => selectAccounts(state),
  'should return AccountList pollingProcessID': () =>
    selectPollingProcessID(state),
  'should return AccountList autoUploadEnabled': () =>
    selectAutoUploadEnabled(state),
  'should return AccountList hostObfuscationEnabled': () =>
    selectHostObfuscationEnabled(state),
};

describe('AccountList selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
