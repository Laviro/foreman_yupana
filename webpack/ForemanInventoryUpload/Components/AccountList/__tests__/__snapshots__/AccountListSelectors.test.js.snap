// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList selectors should return AccountList accounts 1`] = `
Object {
  "Account1": Object {
    "generate_report_status": "running",
    "id": 1,
    "upload_report_status": "running",
  },
  "Account2": Object {
    "generate_report_status": "failure",
    "id": 2,
    "upload_report_status": "unknown",
  },
  "Account3": Object {
    "generate_report_status": "running",
    "id": 3,
    "upload_report_status": "success",
  },
}
`;

exports[`AccountList selectors should return AccountList autoUploadEnabled 1`] = `true`;

exports[`AccountList selectors should return AccountList cloudToken 1`] = `true`;

exports[`AccountList selectors should return AccountList excludePackages 1`] = `false`;

exports[`AccountList selectors should return AccountList hostObfuscationEnabled 1`] = `true`;

exports[`AccountList selectors should return AccountList pollingProcessID 1`] = `0`;

exports[`AccountList selectors should return AccountsList 1`] = `
Object {
  "accounts": Object {
    "Account1": Object {
      "generate_report_status": "running",
      "id": 1,
      "upload_report_status": "running",
    },
    "Account2": Object {
      "generate_report_status": "failure",
      "id": 2,
      "upload_report_status": "unknown",
    },
    "Account3": Object {
      "generate_report_status": "running",
      "id": 3,
      "upload_report_status": "success",
    },
  },
  "autoUploadEnabled": true,
  "cloudToken": true,
  "excludePackages": false,
  "hostObfuscationEnabled": true,
  "pollingProcessID": 0,
}
`;
